widget/embedded/src/pages/Home.tsx

```diff
@@ -57,10 +58,27 @@ export function Home() {
     inputUsdValue,
     outputAmount,
     outputUsdValue,
-    quote,
+    selectedQuote,
+    refetchQuote,
+    quotes,
     resetQuoteWallets,
     setQuoteWarningsConfirmed,
+    setSelectedQuote,
+    setSelectedTag,
+    setRefetchQuote,
   } = useQuoteStore();
+  const {
+    fetch: fetchQuote,
+    loading,
+    error: quoteError,
+    warning: quoteWarning,
+  } = useSwapInput({ refetchQuote });
+
+  // Extract tags from each object and flatten them into a single array
+  const allTags = quotes?.results.flatMap((item) => item.tags);
+  const uniqueTagsArray = sortTags(getUniqueTags(allTags || [])).filter(
+    (tag) => tag.value !== 'HIGH_IMPACT' && tag.value !== 'CENTRALIZED'
+  );
```

Comment:
I believe using just "uniqueTags" is sufficient because its type provides information about the structure of the variables.
You also have the option to transfer additional logic to helper functions.